src/components/tx/SignOrExecuteForm/hooks.ts

```diff
@@ -179,3 +179,10 @@ export const useSafeTxGas = (safeTx: SafeTransaction | undefined): number | unde
 
   return safeTxGas
 }
+
+export const useAlreadySigned = (safeTx: SafeTransaction | undefined): boolean => {
+  const wallet = useWallet()
+  const hasSigned =
+    safeTx && wallet && (safeTx.signatures.has(wallet.address.toLowerCase()) || safeTx.signatures.has(wallet.address))
```

Comment:
What do you think about using `sameAddress` here, e.g.

```suggestion
    safeTx && wallet && Array.from(safeTx.signatures.values()).some((address) => sameAddress(address, wallet.address))
```